resources/js/Hooks/useCustomFieldVisibility.ts

```
@@ -0,0 +1,54 @@
+
+/**
+ * Hook to manage custom field visibility
+ * Watches form values and evaluates visibility rules
+ */
+export function useCustomFieldVisibility({
+    fields,
+    form,
+    namePrefix = 'custom_fields_data',
+}: UseCustomFieldVisibilityOptions) {
+    // Watch all custom field values - only if form instance exists
+    const allFieldValues = form ? (Form.useWatch(namePrefix, form) || {}) : {};
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)